src/main/scala/chisel3/util/Counter.scala

```diff
@@ -67,4 +90,19 @@ object Counter
     when (cond) { wrap := c.inc() }
     (c.value, wrap)
   }
+
+  /** Creates a counter that steps through a specified range of values when a condition is met.
+    *
+    * @param r the range of counter values
+    * @param cond condition that controls whether the counter increments this cycle
+    * @return tuple of the counter value and whether the counter will wrap (the value is at
+    * maximum and the condition is true).
+    */
+  @chiselName
+  def apply(r: Range, cond: Bool = true.B): (UInt, Bool) = {
+    val c = new Counter(r)
+    val wrap = WireInit(false.B)
+    when (cond) { wrap := c.inc() }
+    (c.value, wrap)
```

Comment:
This is a minor API question, but the range API is a little inconsistent with the old API, if you want an enable its:
```scala
Counter(0 until 16, en)
// vs.
Counter(en, 16)
```
I won't contest that maybe it's the existing API that's awkward, but we probably ought to be consistent here

Reply:
@jackkoenig Thanks for your thoughts :thinking: 

Unfortunately, flipping the order of the params would make having default values for the `enable` param more awkward. IMHO, since you can't construct a counter without a range value (i.e. what good is a counter with an `enable` and no range?), it makes sense that this is the first parameter.

My vote would be to deprecate the old API. That way users could have:

```scala
Counter(10)
Counter(10, en)
Counter(0 to 10)
Counter(0 to 10, en)
```

I'm happy to open a new PR to discuss?